### DISCUSSION TITLE
Load API is not behaving as expected

### DISCUSSION THREAD

**Discussion by shaiktas** [milvus-io/milvus#28105] [Q&A and General discussion] [answered]
Milvus Version:  2.3.1
Exeucuting below operation on milvus Client using JAVA Client
1. CreatedCollection with 4 fields
2. Inserted 10 Fields in collection 
3. CreatedIndex
4. Loading Collection
 The Load collection API is struck and after some time it finishing, but load is not performed. From Attu UI it intitially displyed as 50%, after Load API finish execution, it it visiable as unloaded. 

Attaching queryNode logs for reference:
[milvus-querynodelogs-loadAPI.txt](https://github.com/milvus-io/milvus/files/13234115/milvus-querynodelogs-loadAPI.txt)

Attu UI.

<img width="1329" alt="Screenshot 2023-11-01 at 11 00 00 PM" src="https://github.com/milvus-io/milvus/assets/52338042/dc6ec295-3c71-4f52-ad20-c6efc9d99ab5">
<img width="1329" alt="Screenshot 2023-11-01 at 11 00 16 PM" src="https://github.com/milvus-io/milvus/assets/52338042/9f96da55-4d24-41fd-9304-05dd5755a860">

**Comment by shaiktas** [comment]
[javaClientAPI.txt](https://github.com/milvus-io/milvus/files/13234122/javaClientAPI.txt)

**Reply by xiaofan-luan** [reply]
could you offer detailed logs on milvus querynode and querycoord?

**Reply by shaiktas** [reply]
@xiaofan-luan thanks for acknowledgment, i have already attached querynode logs, but here again attaching load API logs from querynode and query coordinator.
[milvus-querycoorlogs-loadAPI-0211.txt](https://github.com/milvus-io/milvus/files/13240874/milvus-querycoorlogs-loadAPI-0211.txt)
[milvus-querynodelogs-loadAPI-0211.txt](https://github.com/milvus-io/milvus/files/13240881/milvus-querynodelogs-loadAPI-0211.txt)
ATTU UI:
<img width="1427" alt="Screenshot 2023-11-02 at 9 27 02 AM" src="https://github.com/milvus-io/milvus/assets/52338042/3fd1e183-3014-4d72-b849-046c1241a4a5">
JavaClient API : 
[javaClientAPI.txt](https://github.com/milvus-io/milvus/files/13240887/javaClientAPI.txt)

**Reply by shaiktas** [reply]
Hi @xiaofan-luan can you please help us when we can expect this log message?
[2023/11/02 14:21:10.145 +00:00] [INFO] [segments/segment_loader.go:436] ["wait segment loaded..."]

**Reply by shaiktas** [reply]
Executed  load API in second attempt. Here are the querycoord, querynode and cluster logs for reference. Same behavior observed.
[milvus-querycoord-load-1102.txt](https://github.com/milvus-io/milvus/files/13246706/milvus-querycoord-load-1102.txt)
[milvus-clusterlogs-1102.txt](https://github.com/milvus-io/milvus/files/13246707/milvus-clusterlogs-1102.txt)
[milvus-querycoord-load-1102.txt](https://github.com/milvus-io/milvus/files/13246709/milvus-querycoord-load-1102.txt)

@xiaofan-luan we are badly blocked since a week with this issue. Appreciate if you can suggest a direction how to debug? 
Below is query node resources:

```
  replicas: 3
  resources:
    limits:
      cpu: 4
      memory: 8Gi
    requests:
      cpu: 2
      memory: 4Gi
```
query coordinator resources:
```
  replicas: 1  
  resources:
    limits:
      cpu: 1
      memory: 2Gi
    requests:
      cpu: 1
      memory: 2Gi
```

**Comment by zhagnlu** [comment]
Could you using this tool https://github.com/zhagnlu/milvus-store-tool to check whether connect and read  storage is ok ?

**Reply by shaiktas** [reply]
@zhagnlu i am not able to build this project. is it on local or any cloud machine? when i am trying to follow up the given instruction in my local mac, i am getting issues: Looks some pre requisites are not matching with this project? 
```
gcc --version
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: arm64-apple-darwin22.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```
```
cmake --version
cmake version 3.27.7

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```
```
 conan --version
Conan version 1.58.0
```
```
make install
Installing dependencies with Conan...
cd /Users/mahaboobbashashaik/work/gitrepos/milvus-isds-bigdata/local/Milvus/milvus-tools/milvus-store-tool/build && conan install ..  --build=missing -s compiler=clang -s compiler.libcxx=libc++ -s compiler.cppstd=17 || { echo 'conan install failed'; exit 1; }
ERROR: The specified 'compiler.cppstd=17' is not available for 'clang None'. Possible values are []'
conan install failed
make: *** [conan_install] Error 1
```

**Reply by xiaofan-luan** [reply]
@jiaoew1991 please help on it

**Reply by shaiktas** [reply]
Hello, i understood that this tool is being recommended `https://github.com/zhagnlu/milvus-store-tool ` to test/verify the storage  connectivity, When i perform insert operation i am able to see the storage (object storage) size is keep increasing .  We have also observed that when bucket size is reached its max capacity, data node keep crashing . So this indicates storage connectivity is success.

**Reply by xiaofan-luan** [reply]
If bucket size has a limitation in your environemnt, 
I would recommend you to enable disk capacity protection.
On the otherside, we would expect your Minio capacity is at least 3-4 times of your raw data size.

**Reply by shaiktas** [reply]
Thanks @xiaofan-luan , yes we do have storage much higher than the raw data size.  I was just trying to share you that the storage connectivity is not an issue. The only blocker is LOAD API is not functioning as expected, and we are looking some pointers here how to debug and troubleshoot this.

**Reply by shaiktas** [reply]
Hi @xiaofan-luan i have attached queryNode, queryCoord logs for load API, also i am seeing below WARN from rootcoordinator: Is it a symptom of failing load API?
`

> [2023/11/05 20:51:56.686 +00:00] [WARN] [rootcoord/quota_center.go:742] ["failed to get collection rate limit config"] [collectionID=445345169243756901] [error="collection=445345169243756901: collection not found"] [errorVerbose="collection=445345169243756901: collection not found\n(1) attached stack trace\n  -- stack trace:\n  | github.com/milvus-io/milvus/pkg/util/merr.wrapWithField\n  | \t/go/src/github.com/milvus-io/milvus/pkg/util/merr/utils.go:531\n  | github.com/milvus-io/milvus/pkg/util/merr.WrapErrCollectionNotFound\n  | \t/go/src/github.com/milvus-io/milvus/pkg/util/merr/utils.go:242\n  | github.com/milvus-io/milvus/internal/rootcoord.(*MetaTable).getLatestCollectionByIDInternal\n  | \t/go/src/github.com/milvus-io/milvus/internal/rootcoord/meta_table.go:503\n  | github.com/milvus-io/milvus/internal/rootcoord.(*MetaTable).getCollectionByIDInternal\n  | \t/go/src/github.com/milvus-io/milvus/internal/rootcoord/meta_table.go:511\n  | github.com/milvus-io/milvus/internal/rootcoord.(*MetaTable).GetCollectionByID\n  | \t/go/src/github.com/milvus-io/milvus/internal/rootcoord/meta_table.go:595\n  | github.com/milvus-io/milvus/internal/rootcoord.(*QuotaCenter).getCollectionLimitConfig\n  | \t/go/src/github.com/milvus-io/milvus/internal/rootcoord/quota_center.go:740\n  | github.com/milvus-io/milvus/internal/rootcoord.(*QuotaCenter).checkDiskQuota\n  | \t/go/src/github.com/milvus-io/milvus/internal/rootcoord/quota_center.go:769\n  | github.com/milvus-io/milvus/internal/rootcoord.(*QuotaCenter).calculateWriteRates\n  | \t/go/src/github.com/milvus-io/milvus/internal/rootcoord/quota_center.go:431\n  | github.com/milvus-io/milvus/internal/rootcoord.(*QuotaCenter).calculateRates\n  | \t/go/src/github.com/milvus-io/milvus/internal/rootcoord/quota_center.go:680\n  | github.com/milvus-io/milvus/internal/rootcoord.(*QuotaCenter).run\n  | \t/go/src/github.com/milvus-io/milvus/internal/rootcoord/quota_center.go:149\n  | runtime.goexit\n  | \t/usr/local/go/src/runtime/asm_amd64.s:1598\nWraps: (2) collection=445345169243756901\nWraps: (3) collection not found\nError types: (1) *withstack.withStack (2) *errutil.withPrefix (3) merr.milvusError"]

`

**Reply by xiaofan-luan** [reply]
@shaikatas
seems that you don't have enough local disk.
are you trying to use diskANN?
then you need to config your data dir for cache local data.
The local data dir must use NVME disk

**Reply by shaiktas** [reply]
Thanks @xiaofan-luan , i am using helm for Milvus and its dependencies deployment.  Please help to understand which configs will confirm this `are you trying to use diskANN?`? Below is the output of describe pod command of rootcoordinator.

```
`kubectl describe pod milvus-int-rootcoord-9f74bd4d6-96mwd -n opsmx
Name:             milvus-int-rootcoord-9f74bd4d6-96mwd
Namespace:        opsmx
Priority:         0
Service Account:  default
Node:             is-milvus-cluster-default-6889c7b6b4xlwbdf-ss568/17.182.27.109
Start Time:       Sun, 05 Nov 2023 21:56:50 -0600
Labels:           app.kubernetes.io/instance=milvus-int
                  app.kubernetes.io/managed-by=spinnaker
                  app.kubernetes.io/name=milvus
                  component=rootcoord
                  pod-template-hash=9f74bd4d6
Annotations:      artifact.spinnaker.io/location: 
                  artifact.spinnaker.io/name: milvus-int-rootcoord
                  artifact.spinnaker.io/type: kubernetes/deployment
                  artifact.spinnaker.io/version: 
                  checksum/config: /configmap.yaml
                  moniker.spinnaker.io/application: milvus
                  moniker.spinnaker.io/cluster: deployment milvus-int-rootcoord
Status:           Running
IP:               192.168.14.234
IPs:
  IP:           192.168.14.234
Controlled By:  ReplicaSet/milvus-int-rootcoord-9f74bd4d6
Init Containers:
  config:
    Container ID:  containerd://f2e066005f86716c0db394ee76b96f5186c429832955667f4d440ae81264c4d9
    Image:         docker-upstream.apple.com/milvusdb/milvus-config-tool:v0.1.1
    Image ID:      docker-upstream.apple.com/milvusdb/milvus-config-tool@sha256:e407880f9af0f38afa9412b5e7100f9b99d2f3eb3cacc10e1a12f06070725745
    Port:          <none>
    Host Port:     <none>
    Command:
      /cp
      /run-helm.sh,/merge
      /milvus/tools/run-helm.sh,/milvus/tools/merge
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Sun, 05 Nov 2023 21:56:51 -0600
      Finished:     Sun, 05 Nov 2023 21:56:51 -0600
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /milvus/tools from tools (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-58mkz (ro)
Containers:
  rootcoord:
    Container ID:  containerd://99fb0bd0970d786d0d435516658029f8ce6b6efac533272ce359d2e22e3592f6
    Image:         docker-upstream.apple.com/milvusdb/milvus:v2.3.1
    Image ID:      docker-upstream.apple.com/milvusdb/milvus@sha256:9f37c7100c44be8e7419c55f285c2963ee4d25c71951a2325614a4565a03d7e6
    Ports:         53100/TCP, 9091/TCP
    Host Ports:    0/TCP, 0/TCP
    Args:
      /milvus/tools/run-helm.sh
      milvus
      run
      rootcoord
    State:          Running
      Started:      Sun, 05 Nov 2023 21:56:52 -0600
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     2
      memory:  2Gi
    Requests:
      cpu:        2
      memory:     2Gi
    Liveness:     http-get http://:metrics/healthz delay=90s timeout=5s period=30s #success=1 #failure=5
    Readiness:    http-get http://:metrics/healthz delay=90s timeout=5s period=10s #success=1 #failure=5
    Environment:  <none>
    Mounts:
      /milvus/configs/default.yaml from milvus-config (ro,path="default.yaml")
      /milvus/configs/user.yaml from milvus-config (ro,path="user.yaml")
      /milvus/tools from tools (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-58mkz (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  milvus-config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      milvus-int-v004
    Optional:  false
  tools:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
  kube-api-access-58mkz:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
```

**Reply by shaiktas** [reply]
Also @xiaofan-luan , i see below error logs from index coordinator, same objectstore is being connected successfully from data node, but it is failing form index coordinator. any clue why is working form one node and not in other? 

> [2023/11/06 04:04:57.652 +00:00] [DEBUG] [indexnode/task_scheduler.go:219] ["process task"] [task=by-dev/445440313142792561]
> [2023/11/06 04:04:57.652 +00:00] [INFO] [indexnode/task.go:153] ["Begin to prepare indexBuildTask"] [buildID=445440313142792561] [Collection=0] [SegmentID=0]
> [2023/11/06 04:04:57.652 +00:00] [INFO] [indexnode/task.go:181] ["Successfully prepare indexBuildTask"] [buildID=445440313142792561] [Collection=0] [SegmentID=0]
> [2023/11/06 04:04:57.761 +00:00] [INFO] [indexnode/task.go:322] ["index params are ready"] [buildID=445440313142792561] ["index params"="{\"dim\":\"3\",\"index_type\":\"IVF_FLAT\",\"metric_type\":\"L2\",\"nlist\":\"256\"}"]
> [ERROR] 2023-11-06 04:04:33.007 AWSClient [137777924527872] HTTP response code: -1
> Resolved remote host IP address: 
> Request ID: 
> Exception name: 
> Error message: curlCode: 7, Couldn't connect to server
> 0 response headers:
> I20231106 04:04:36.210685    54 MinioChunkManager.cpp:96] [SEGCORE][ProcessFormattedStatement][milvus] [AWS LOG] [ERROR] 2023-11-06 04:04:36.210 CurlHttpClient [137777924527872] Curl returned error code 7 - Couldn't connect to server
> I20231106 04:04:36.210757    54 MinioChunkManager.cpp:96] [SEGCORE][ProcessFormattedStatement][milvus] [AWS LOG] [ERROR] 2023-11-06 04:04:36.210 AWSClient [137777924527872] HTTP response code: -1
> Resolved remote host IP address: 
> Request ID: 
> Exception name: 
> Error message: curlCode: 7, Couldn't connect to server
> 0 response headers:
> I20231106 04:04:42.649242    54 MinioChunkManager.cpp:96] [SEGCORE][ProcessFormattedStatement][milvus] [AWS LOG] [ERROR] 2023-11-06 04:04:42.649 CurlHttpClient [137777924527872] Curl returned error code 7 - Couldn't connect to server
> I20231106 04:04:42.649341    54 MinioChunkManager.cpp:96] [SEGCORE][ProcessFormattedStatement][milvus] [AWS LOG] [ERROR] 2023-11-06 04:04:42.649 AWSClient [137777924527872] HTTP response code: -1
> Resolved remote host IP address: 
> Request ID: 
> Exception name: 
> Error message: curlCode: 7, Couldn't connect to server
> 0 response headers:
> I20231106 04:04:55.522343    54 MinioChunkManager.cpp:96] [SEGCORE][ProcessFormattedStatement][milvus] [AWS LOG] [ERROR] 2023-11-06 04:04:55.522 CurlHttpClient [137777924527872] Curl returned error code 7 - Couldn't connect to server
> I20231106 04:04:55.522456    54 MinioChunkManager.cpp:96] [SEGCORE][ProcessFormattedStatement][milvus] [AWS LOG] [ERROR] 2023-11-06 04:04:55.522 AWSClient [137777924527872] HTTP response code: -1
> Resolved remote host IP address: 
> Request ID: 
> Exception name: 
> Error message: curlCode: 7, Couldn't connect to server
> 0 response headers:
> I20231106 04:04:57.761328    44 MinioChunkManager.cpp:125] [SEGCORE][InitSDKAPI][milvus] init aws with log level:error
> I20231106 04:04:57.767640    44 MinioChunkManager.cpp:285] [SEGCORE][MinioChunkManager][milvus] init MinioChunkManager with parameter[endpoint: 'xxxxxx.objectstore.is.corp.xxxxcom:', default_bucket_name:'isds-milvus-storage', use_secure:'false']
> I20231106 04:04:57.767691    44 factory.cc:20] [KNOWHERE][Create][milvus] create knowhere index IVF_FLAT
> I20231106 04:04:57.770606    52 MinioChunkManager.cpp:96] [SEGCORE][ProcessFormattedStatement][milvus] [AWS LOG] [ERROR] 2023-11-06 04:04:57.770 CurlHttpClient [137777941313280] Curl returned error code 7 - Couldn't connect to server
> I20231106 04:04:57.770700    52 MinioChunkManager.cpp:96] [SEGCORE][ProcessFormattedStatement][milvus] [AWS LOG] [ERROR] 2023-11-06 04:04:57.770 AWSClient [137777941313280] HTTP response code: -1
> Resolved remote host IP address: 
> Request ID: 
> Exception name: 
> Error message: curlCode: 7, Couldn't connect to server

**Reply by shaiktas** [reply]
Hello @yiwangdr , @zhagnlu @xiaofan-luan , from below querycoord logs  `[task/scheduler.go:558] ["processed tasks"] [nodeID=206] [toProcessNum=1] [failCount=1] [toRemoveNum=0]`  , from where it is seeks faiCount=1? it should be zero correct?  Any way to corelate this failCount value from other svc or depedencies  (Etcd/Pulsar/ logs? 

```
1. [2023/11/07 04:08:35.408 +00:00] [INFO] [task/executor.go:183] ["load segments..."] [taskIDs="[1699286969416]"] [collectionID=445457864570910134] [replicaID=445457883303772166] [shard=by-dev-rootcoord-dml_8_445457864570910134v0] [segmentIDs="[445457864570910146]"] [nodeID=206] [source=1]
2. [2023/11/07 04:08:35.411 +00:00] [DEBUG] [task/scheduler.go:523] ["process tasks related to node"] [nodeID=206] [processingTaskNum=1] [waitingTaskNum=0] [segmentTaskNum=1] [channelTaskNum=0]
3. [2023/11/07 04:08:35.411 +00:00] [DEBUG] [funcutil/parallel.go:54] [process] [total=1] ["time cost"=46.038µs]
4. [2023/11/07 04:08:35.411 +00:00] [INFO] [task/scheduler.go:558] ["processed tasks"] [nodeID=206] [toProcessNum=1] [failCount=1] [toRemoveNum=0]
5. [2023/11/07 04:08:35.411 +00:00] [INFO] [task/scheduler.go:564] ["process tasks related to node done"] [nodeID=206] [processingTaskNum=1] [waitingTaskNum=0] [segmentTaskNum=1] [channelTaskNum=0]
6. [2023/11/07 04:08:35.418 +00:00] [INFO] [observers/collection_observer.go:122] ["observes partitions timeout"] [partitionNum=1]
7. [2023/11/07 04:08:35.418 +00:00] [INFO] [observers/collection_observer.go:159] ["observe partitions status"] [partitionNum=1]
8. [2023/11/07 04:08:35.418 +00:00] [INFO] [observers/collection_observer.go:193] ["partition targets"] [collectionID=445457864570910134] [partitionID=445457864570910135] [segmentTargetNum=1] [channelTargetNum=1] [totalTargetNum=2] [replicaNum=1]
9. [2023/11/07 04:08:35.418 +00:00] [INFO] [observers/collection_observer.go:216] ["partition load progress"] [collectionID=445457864570910134] [partitionID=445457864570910135] [subChannelCount=1] [loadSegmentCount=0]
10. [2023/11/07 04:08:35.418 +00:00] [INFO] [balance/utils.go:70] ["create segment task"] [collection=445457864570910134] [segmentID=445457864570910146] [replica=445457883303772166] [channel=by-dev-rootcoord-dml_8_445457864570910134v0] [from=-1] [to=206]
11. [2023/11/07 04:08:35.886 +00:00] [INFO] [querycoordv2/services.go:58] ["show collections request received"] [traceID=10046b596db9e40902a6101a72f50d77] [collections="[]"]
12. [2023/11/07 04:08:35.898 +00:00] [INFO] [querycoordv2/services.go:787] ["get replicas request received"] [traceID=d6184205ef00c48808557ca97a309dd8] [collectionID=445457864570910134] [with-shard-nodes=false]
13. [2023/11/07 04:08:35.898 +00:00] [WARN] [querycoordv2/handlers.go:321] ["failed to get channels, collection not loaded"]
14. [2023/11/07 04:08:35.898 +00:00] [WARN] [querycoordv2/services.go:824] ["failed to get replica info"] [traceID=d6184205ef00c48808557ca97a309dd8] [collectionID=445457864570910134] [replica=445457883303772166] [error="failed to get channels, collection not loaded: collection=445457864570910134: collection not found"] [errorVerbose="failed to get channels, collection not loaded: collection=445457864570910134: collection not found\n(1) attached stack trace\n  -- stack trace:\n  | github.com/milvus-io/milvus/pkg/util/merr.WrapErrCollectionNotFound\n  | \t/go/src/github.com/milvus-io/milvus/pkg/util/merr/utils.go:244\n  | [...repeated from below...]\nWraps: (2) failed to get channels, collection not loaded\nWraps: (3) attached stack trace\n  -- stack trace:\n  | github.com/milvus-io/milvus/pkg/util/merr.wrapWithField\n  | \t/go/src/github.com/milvus-io/milvus/pkg/util/merr/utils.go:531\n  | github.com/milvus-io/milvus/pkg/util/merr.WrapErrCollectionNotFound\n  | \t/go/src/github.com/milvus-io/milvus/pkg/util/merr/utils.go:242\n  | github.com/milvus-io/milvus/internal/querycoordv2.(*Server).fillReplicaInfo\n  | \t/go/src/github.com/milvus-io/milvus/internal/querycoordv2/handlers.go:322\n  | github.com/milvus-io/milvus/internal/querycoordv2.(*Server).GetReplicas\n  | \t/go/src/github.com/milvus-io/milvus/internal/querycoordv2/services.go:822\n  | github.com/milvus-io/milvus/internal/distributed/querycoord.(*Server).GetReplicas\n  | \t/go/src/github.com/milvus-io/milvus/internal/distributed/querycoord/service.go:379\n  | github.com/milvus-io/milvus/internal/proto/querypb._QueryCoord_GetReplicas_Handler.func1\n  | \t/go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:5537\n  | github.com/milvus-io/milvus/pkg/util/interceptor.ServerIDValidationUnaryServerInterceptor.func1\n  | \t/go/src/github.com/milvus-io/milvus/pkg/util/interceptor/server_id_interceptor.go:54\n  | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1\n  | \t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25\n  | github.com/milvus-io/milvus/pkg/util/interceptor.ClusterValidationUnaryServerInterceptor.func1\n  | \t/go/src/github.com/milvus-io/milvus/pkg/util/interceptor/cluster_interceptor.go:48\n  | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1\n  | \t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25\n  | github.com/milvus-io/milvus/pkg/util/logutil.UnaryTraceLoggerInterceptor\n  | \t/go/src/github.com/milvus-io/milvus/pkg/util/logutil/grpc_interceptor.go:23\n  | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1\n  | \t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25\n  | go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryServerInterceptor.func1\n  | \t/go/pkg/mod/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.38.0/interceptor.go:342\n  | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1\n  | \t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25\n  | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1\n  | \t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34\n  | github.com/milvus-io/milvus/internal/proto/querypb._QueryCoord_GetReplicas_Handler\n  | \t/go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:5539\n  | google.golang.org/grpc.(*Server).processUnaryRPC\n  | \t/go/pkg/mod/google.golang.org/grpc@v1.54.0/server.go:1345\n  | google.golang.org/grpc.(*Server).handleStream\n  | \t/go/pkg/mod/google.golang.org/grpc@v1.54.0/server.go:1722\n  | google.golang.org/grpc.(*Server).serveStreams.func1.2\n  | \t/go/pkg/mod/google.golang.org/grpc@v1.54.0/server.go:966\n  | runtime.goexit\n  | \t/usr/local/go/src/runtime/asm_amd64.s:1598\nWraps: (4) collection=445457864570910134\nWraps: (5) collection not found\nError types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) merr.milvusError"]
15. [2023/11/07 04:08:35.912 +00:00] [DEBUG] [task/scheduler.go:523] ["process tasks related to node"] [nodeID=206] [processingTaskNum=1] [waitingTaskNum=0] [segmentTaskNum=1] [channelTaskNum=0]
```

**Reply by shaiktas** [reply]
Hello Team/ @zhagnlu @xiaofan-luan  can we expect some response please from any one? we are blocked with this since more than 2 weeks, we are deploying through helm charts  (https://github.com/zilliztech/milvus-helm/tree/master/charts/milvus). Please go through above source of provided information , attached logs.
 Our source of Deployment is : https://github.com/zilliztech/milvus-helm/tree/master/charts/milvus

**Reply by xiaofan-luan** [reply]
Sorry for the late update.
I'm checking the log but there seems to be no clear clue.

1. What kind of storage did you use? Is is S3 or Minio？How did you deploy your S3
2. Did you try to https://github.com/zhagnlu/milvus-store-tool and check your blob storage can be accessed?

@zhagnlu any other advice?

**Reply by xiaofan-luan** [reply]
would like to offer more help.
Please contact me at xiaofan.luan@zilliz.com and we would like to offer more help

**Reply by locustbaby** [reply]
@shaiktas https://github.com/milvus-io/milvus/blob/master/deployments/export-log/README.md
Can you try this tool and share all your logs with us？

**Comment by shaiktas** [comment]
Thanks for your email.
I am working on it to provide you requested information.
For your information, when i install milvus locally through docker-compose,
it works fine, Also it works file with milvus-operator on Cloud Platform.
Issue is only with helm charts.  (https://github.com/zilliztech/milvus-helm)

By the way, What is "milvus-config-tool"?? is is mounting milvus.yaml under
/milvus/config/ path?

Regards
Mahaboob..

**Reply by shaiktas** [reply]
Hi @xiaofan-luan here i am attaching logs of API's starting from createCollection to till load API failure followed by search API.  We are trying to execute GeneralExample from (https://github.com/milvus-io/milvus-sdk-java/blob/master/examples/main/java/io/milvus/GeneralExample.java). Same test cases ran successfully with docker-compose locally installed milvus , issue is only with helm charts deployment (https://github.com/zilliztech/milvus-helm/tree/master/charts).
[milvus-standalone-mode.logs.zip](https://github.com/milvus-io/milvus/files/13303817/milvus-standalone-mode.logs.zip)

**Reply by locustbaby** [reply]
`milvus-config-tool` is container inside of milvus' pod which will help merge custom config into milvus.
And also it will be help if you can share you values of your charts deploy.

**Reply by shaiktas** [reply]
is `milvus-config-tool` is generating milvus.yaml file under /milvus/config??
Here is my helm values yaml for reference: 
[helmvalues.yaml.zip](https://github.com/milvus-io/milvus/files/13304197/helmvalues.yaml.zip)

**Reply by LoveEachDay** [reply]
@shaiktas `milvus-config-tool` will take `user.yaml` and `milvus.yaml` as input, then merge the config back to `milvus.yaml`.

**Reply by locustbaby** [reply]
Thanks for your information, wanna to make a confirm:
1. In your values, externalS3.port and externalS3.rootPath is `""`,is that real config or just for desensitize.
2. When you use docker-compose, are you also using object storage in values with same config, or default minio?

**Reply by locustbaby** [reply, marked as answer]
@shaiktas Would like to continue this in slack channel

**Reply by shaiktas** [reply]
When i am using docker-compose, i am using default provided minio within docker-compose, no edits, no customization it is used as it is.. `wget https://github.com/milvus-io/milvus/releases/download/v2.3.1/milvus-standalone-docker-compose.yml -O docker-compose.yml`

**Reply by shaiktas** [reply]
@locustbaby ,  externalS3.port and externalS3.rootPath configs are "" in my case, and  with these configs looks insert operation successful, as i see increase bucket size from 0 to 817 KB (0.1% )after inserting 100 entries. 
<img width="563" alt="Screenshot 2023-11-09 at 11 39 15 AM" src="https://github.com/milvus-io/milvus/assets/52338042/3de8fcec-d616-428a-b087-f9565e4f16a5">

**Reply by locustbaby** [reply]
@shaiktas Can we continue this issue in Slack?

**Reply by shaiktas** [reply]
Thanks @locustbaby , here is slack channel id : https://milvusio.slack.com/archives/C05AV3W81NK/p1699512530977979

**Comment by xiaofan-luan** [comment]
We've find one bug while loading https://github.com/milvus-io/milvus/issues/28303
will release 2.3.3 to fix the issue.

This bug is triggered when S3 hit error.
So we still need to investigate on why S3 hit error

**Reply by shaiktas** [reply]
Thank you @xiaofan-luan for hinting that S3 issue, the issue seems due to below externalS3 configs:        After setting as below, load API behaving as expected and proceeding further with search...                                                         
   `port: "80",  rootPath: "files"`

**Reply by locustbaby** [reply]
@shaiktas We've located the bug that cpp's SDK won't set 80 as default port when port is empty, but golang's  SDK will, will be fixed soon.
